packages/flow/lib/AstroCase.d.ts

```diff
@@ -1,21 +1,17 @@
-export type Props<T> = 'of' extends keyof T
-	? T['of'] extends (string | number | bigint | boolean | symbol | null | undefined)
-		? {
-			of: T
-		}
-	: T['of'] extends ((value: unknown) => boolean)
-		? {
-			of: T
-		}
-	: T['of'] extends object
-		? {
-			of: T
-		}
-	: never
-: 'default' extends keyof T
-	? {
-		default: "" | true
-	}
-: any
 
-export default function Case<T>(props: Props<T>): any
+type AllowedTypes = string | null | bigint | Record<PropertyKey, unknown> | undefined | number | boolean
```

Comment:
If you don't know `Record<PropertyKey, unknown>` stands for the typical object literals that we type every day. The point of allowed types is to just be explicit. when it comes to writing an object this is the right way to express this intent.